test/compare_outputs.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python3
+
+"""
+Script to compare the output of two PyBDSF runs. The generated FITS images,
+FITS catalog, and text files will be compared.
+"""
+
+import argparse
+import filecmp
+import logging
+import os
+import sys
+import textwrap
+import warnings
+
+import astropy.io.fits
+import numpy as np
+import difflib
+
+import astropy.wcs
+import re
+
+
+# Suppress UserWarning messages (mostly generated by matplotlib) and
+# FITSFixedWarning messages (generated by harmless WCS header issues)
+warnings.filterwarnings("ignore", category=UserWarning)
+warnings.filterwarnings("ignore", category=astropy.wcs.FITSFixedWarning)
+
+# Global Logger object, will be initialized in init_logger()
+logger = None
+
+
+def check_all_files_present(dcmp):
+    """
+    Checks that all files are present in both compared directories.
+
+    :param dircmp dcmp: the result of `dircmp(left, right)`
+    :return bool: True if all files are present, False if not
+    """
+    logger.debug("Comparing directory contents of '%s' and '%s'", dcmp.left, dcmp.right)
+    agree = True
+    if dcmp.left_only or dcmp.right_only:
+        agree = False
+        msg = "The following files are present in '%s' but not in '%s': %s"
+        if dcmp.left_only:
+            logger.error(msg, dcmp.left, dcmp.right, dcmp.left_only)
+        if dcmp.right_only:
+            logger.error(msg, dcmp.right, dcmp.left, dcmp.right_only)
+    return agree
+
+
+def text_files_are_similar(file1, file2, verbosity):
+    """
+    Compare text files ``file1`` and ``file2``, using `difflib`.
+    :param str file1: First text file
+    :param str file2: Second text file
+    :param int verbosity: Verbosity level, higher is more verbose
+    :return bool: True if text files are similar, else False
+    """
+    logger.debug("Comparing '%s' and '%s'", file1, file2)
+    d = difflib.Differ()
+    with open(file1) as f1, open(file2) as f2:
+        diff = list(d.compare(f1.readlines(), f2.readlines()))
+    agree = True
+    for line in diff:
```

Comment:
I would filter `diff` already now, since you need it both directly below and in the log (line 77). Something like:
```
    # Only keep lines that start with diff markers and are not comment lines
    diff = [
        line for line in d.compare(f1.readlines(), f2.readlines()) 
        if line.startswith(('-','+')) and not line[2:].startswith('#')
    ] 
```

Reply:
Good suggestion -- done.